### PR TITLE
CB-12065: Fix FreeIPA HA repair to update cached kerberos config

### PR DESCRIPTION
**Pull request by jamisonbennett**
Fix FreeIPA HA repair so that the cached kerberos configuration is
updated. Specifically the DNS nameservers were stale which caused
DataLakes and DataHubs to have DNS break after all of the original
FreeIPA servers have been repaired.

This was tested by validating the cached kerberos configuration is
updated during downscale and upscale operations.

See detailed description in the commit message.

### CODE REVIEW

**Review by christmasferret — APPROVED** [review]
LGTM

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/config/KerberosConfigUpdateService.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    @Inject
+    private KerberosConfigService kerberosConfigService;
+
+    @Inject
+    private StackService stackService;
+
+    public KerberosConfig updateNameservers(Long stackId) {
+        Stack stack = getStackWithInstanceMetadata(stackId);
+        String environmentCrn = stack.getEnvironmentCrn();
+        KerberosConfig kerberosConfig = kerberosConfigService.get(environmentCrn);
```
Comment (lacikaaa) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/config/KerberosConfigUpdateService.java:32]: this won't return all. you have to use `kerberosConfigRepository.findByAccountIdAndEnvironmentCrnAndArchivedIsFalse` on repository level, and this is not exposed on service level right now

**Review by lacikaaa — CHANGES_REQUESTED** [review]

**Review by lacikaaa — APPROVED** [review]

**Comment by lacikaaa** [comment]
this LGTM. but there are old LDAP configs, for old FreeIPA installation where we don't have the CNAME. see: `LdapConfigRegisterService#addServerHost`
we might need to handle those if there is any. we should check that before making any changes

**Comment by jamisonbennett** [comment]
@lacikaaa Thanks for mentioning this. `LdapConfigRegisterService#addServerHost` was supported from 2.20 and later. The repair flow only is possible for HA configurations which are much later (2.31?). So this shouldn't be a problem because we can't repair something that old anyway.

**Comment by lacikaaa** [comment]
@jamisonbennett thanks for the clarification
what about upscale, when we are migrating from single to HA in the foreseeable future?

**Pull request merged**